bin/pytorch_inference/CCommandParser.cc

```diff
@@ -281,8 +291,18 @@ CCommandParser::jsonToInferenceRequest(const rapidjson::Document& doc) {
 
 CCommandParser::SControlMessage
 CCommandParser::jsonToControlMessage(const rapidjson::Document& doc) {
-    return {static_cast<EControlMessageType>(doc[CONTROL].GetInt()),
-            doc[NUM_ALLOCATIONS].GetInt(), doc[REQUEST_ID].GetString()};
+    auto controlMessageType = static_cast<EControlMessageType>(doc[CONTROL].GetInt());
+    switch (controlMessageType) {
+    case E_NumberOfAllocations:
+        return {controlMessageType, doc[NUM_ALLOCATIONS].GetInt(),
+                doc[REQUEST_ID].GetString()};
+    case E_ClearCache:
+        return {controlMessageType, 0, doc[REQUEST_ID].GetString()};
+    case E_Unknown:
+        break;
+    }
+
+    LOG_ABORT(<< "Programmatic error - incorrect validation of control message");
```

Comment:
We should never get here as `validateControlMessageJson()` should have responded with an error if the intended methods are called in the correct order.